### PR TITLE
exclude removed reactables

### PR DESCRIPTION
**Pull request by willmcgugan**
Fixes https://github.com/Textualize/textual/issues/1714

When calling `switch_screen` a screen is removed, but the watchers in the header were called after the DOM had been deleted.

- Deprecates reactive.watch in favor of DOMNode.watch
- Removes watch callbacks for detached nodes

### CODE REVIEW

**Comment by willmcgugan** [comment]
Bit premature @davep

**Comment by davep** [comment]
> Bit premature @davep

D'oh! Could have sworn I'd seen a review request come in on this one. Apologies, getting stuff mixed up!

**Comment by willmcgugan** [comment]
No worries. Ready now.

**Comment by davep** [comment]
Hey look, it works now! ;-)

**Review by davep — APPROVED** [review]

File: src/textual/message_pump.py [hunk 1]
```diff
@@ -123,6 +123,20 @@ def log(self) -> Logger:
         """
         return self.app._logger
 
+    @property
+    def is_attached(self) -> bool:
+        """Check the node is attached to the app via the DOM."""
```
[the remark this answers is not in the document]
Comment (rodrigogiraoserrao) [reply src/textual/message_pump.py]: Just the type of thing I would say :')

**Pull request merged**